src/Extensions/StreamContextExtensions.cs

```diff
@@ -1,24 +1,18 @@
-﻿using System.Threading.Tasks;
+﻿using System;
+using System.Threading.Tasks;
 using Akka.Streams.Dsl;
-using Arcane.Stream.BlobStorage.Exceptions;
 using Arcane.Stream.BlobStorage.Models;
-using Snd.Sdk.Storage.Base;
 using Snd.Sdk.Storage.Models.Base;
 
 namespace Arcane.Stream.BlobStorage.Extensions;
 
 public static class StreamContextExtensions
 {
-    public static Sink<(IStoragePath, string), Task> GetSink(this BlobStorageStreamContext context, IBlobStorageService blobStorageService)
+    public static Sink<(IStoragePath, string), Task> GetSink(this BlobStorageStreamContext context, Func<(IStoragePath, string), Task > action)
     {
         return Sink.ForEachAsync<(IStoragePath, string)>(context.DeleteParallelism, async deleteRequest =>
         {
-            var (sourceRoot, sourceBlobName) = deleteRequest;
-            var res = await blobStorageService.RemoveBlob(sourceRoot.ObjectKey, sourceBlobName);
-            if (!res)
-            {
-                throw new SinkException($"Failed to remove blob {sourceBlobName} from {sourceRoot}");
-            }
+            await action(deleteRequest);
```

Comment:
if you do this, do you even need to map materialized value? (ie just return `deleteRequest`?)

Reply:
It's required here because the `ForEachAsync` sink returns `Task<Done` instead of `Task`.